src/engine/TransitivePath.cpp

```diff
@@ -215,361 +218,168 @@ size_t TransitivePath::getCostEstimate() {
-    checkTimeoutHashSet();
-    Id l = sub(i, leftSubCol);
-    Id r = sub(i, rightSubCol);
-    MapIt it = edges.find(l);
-    if (it == edges.end()) {
-      std::shared_ptr<ad_utility::HashSet<Id>> s =
-          std::make_shared<ad_utility::HashSet<Id>>();
-      s->insert(r);
-      edges[l] = s;
-    } else {
-      // If r is not in the vector insert it
-      it->second->insert(r);
-    }
+  } else {
+    LOG(DEBUG) << "TransitivePath result computation..." << std::endl;
+    CALL_FIXED_SIZE((std::array{_resultWidth, subWidth}),
+                    &TransitivePath::computeTransitivePath, this, &idTable,
+                    subRes->idTable(), _lhs, _rhs);
   }
 
-  // For every node do a dfs on the graph
-
-  // Stores nodes we already have a path to. This avoids cycles.
-  ad_utility::HashSet<Id> marks;
-
-  // The stack used to store the dfs' progress
-  std::vector<ad_utility::HashSet<Id>::const_iterator> positions;
-  // Used to store all edges leading away from a node for every level.
-  // Reduces access to the hashmap, and is safe as the map will not
-  // be modified after this point.
-  std::vector<std::shared_ptr<const ad_utility::HashSet<Id>>> edgeCache;
+  LOG(DEBUG) << "TransitivePath result computation done." << std::endl;
+  // NOTE: The only place, where the input to a transitive path operation is not
+  // an index scan (which has an empty local vocabulary by default) is the
+  // `LocalVocabTest`. But it doesn't harm to propagate the local vocab here
+  // either.
+  return {std::move(idTable), resultSortedOn(), subRes->getSharedLocalVocab()};
+}
 
-  Id last_elem = ID_NO_VALUE;
-  size_t last_result_begin = 0;
-  size_t last_result_end = 0;
-  for (size_t i = 0; i < left.size(); i++) {
-    checkTimeoutHashSet();
-    if (left[i][leftSideCol] == last_elem) {
-      // We can repeat the last output
-      size_t num_new = last_result_end - last_result_begin;
-      size_t res_row = res.size();
-      res.resize(res.size() + num_new);
-      checkTimeoutAfterNCalls(num_new * resWidth);
-      for (size_t j = 0; j < num_new; j++) {
-        for (size_t c = 0; c < resWidth; c++) {
-          res(res_row + j, c) = res(last_result_begin + j, c);
-        }
-      }
-      continue;
-    }
-    last_elem = left[i][leftSideCol];
-    last_result_begin = res.size();
-    MapIt rootEdges = edges.find(last_elem);
-    if (rootEdges != edges.end()) {
-      positions.push_back(rootEdges->second->begin());
-      edgeCache.push_back(rootEdges->second);
-    }
-    if (minDist == 0) {
-      AD_THROW(
-          "The TransitivePath operation does not support a minimum "
-          "distance of 0 (use at least one instead).");
-    }
+// _____________________________________________________________________________
+std::shared_ptr<TransitivePath> TransitivePath::bindLeftSide(
+    std::shared_ptr<QueryExecutionTree> leftop, size_t inputCol) const {
+  return bindLeftOrRightSide(std::move(leftop), inputCol, true);
```

Comment:
Does the `bindLeftOrRightSide` check that the variables match on the column? It probably should.

Reply:
`bindLeftOrRightSide` does... something if the variables don't match the column. But I'm not exactly sure what it does. Apparently the `ColumnIndex` is adjusted if it doesn't match the `inputCol`. Cf. TransitivePath.cpp:321